packages/components/addon/components/hds/form/character-count/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+export const ID_PREFIX = 'character-count-';
+
+const NOOP = () => {};
+
+export default class HdsFormCharacterCountIndexComponent extends Component {
+  @tracked message;
+  @tracked currentLength; // The current number of characters typed into the associated input
+
+  /**
+   * @param maxLength
+   * @type {number}
+   * @default null
+   * @description The maximum number of characters allowed.
+   */
+  get maxLength() {
+    let { maxLength } = this.args;
+    return maxLength;
+  }
+
+  /**
+   * @param remainingLength
+   * @type {number}
+   * @default null
+   * @description The remaining number of characters.
+   */
+  get remainingLength() {
+    return this.maxLength - this.currentLength;
+  }
```

Comment:
In theory, this can be calculated by the consumer, even in the template by using `ember-math-helpers`, for example `{{sub CC.maxLength CC.currentLegth}}`. However, this helper doesn't seem to be as popular within our consumers' codebase, so instead of forcing them to introduce a new dependency I'd rather do this calculation within the component.

Reply:
Great idea! The remaining count is probably one of the more common use-cases here.

Reply:
I agree with your approach, makes sense 👍